src/nextjournal/clerk/analyzer.clj

```diff
@@ -217,6 +218,21 @@
   (-> (reduce add-block-id (assoc analyzed-doc :blocks [] :id->count {} ) blocks)
       (dissoc :id->count)))
 
+(defn ^:private internal-proxy-name?
+  "Returns true if `sym` represents a var name interned by `clojure.core/proxy`."
+  [sym]
+  (str/includes? (name sym) ".proxy$"))
```

Comment:
I think it's better to make this a `defn` so the arg becomes apparent. Also I've changed the name to be more narrow in order to match the implementation. We can change the implementation when we grow it.